src/boot/z_std_dma.c

```diff
@@ -208,8 +208,12 @@ const char* DmaMgr_GetFileNameImpl(u32 vrom) {
         iter++;
         name++;
     }
+#ifndef AVOID_UB
     //! @bug Since there is no return, in case the file isn't found, the return value will be a pointer to the end
     // of gDmaDataTable
+#else
+    return "";
+#endif
```

Comment:
the intent was to keep the vanilla code always first, and in some cases that means an empty block goes first. Keeping the comments in the block also means that if someone is not interested in ub fixes, they can preprocess the codebase and remove the UB related explanations they arent interested in with it

Reply:
I think this example helps show that intent more
```
#ifndef AVOID_UB
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[nextPlatform]);
    Math_Vec3f_DistXYZ(&player->actor.world.pos, &sPlatformPositions[altNextPlatform]);
#else
    // Defining `AVOID_UB` will remove these function calls entirely as they don't
    // have any effect (and return values are not used).
#endif
```

Reply:
the suggestion was mainly prompted by dragorn, who is coming from the perspective of not even really wanting to introduce UB fixes into the codebase at all. 
Which while i dont agree with entirely, I do understand his perspective. And the point was to be able to remove UB related explanations that someone who is not interested in the fixes wouldnt want to see.

I would personally be ok with making these changes to only use `ifdef` or `ifndef` when it makes sense, but think if theres a situation that needs two blocks that the vanilla code should always be prioritized first

Reply:
about readability I do think using consistently
```c
#ifndef AVOID_UB

#else

#endif
```
would be more readable

then you always get vanilla code first (including nothing, no code)

and you less have to pay close attention to the n in `ifndef` (`ifdef` and `ifndef` being very similar at first glance...)

Currently it's fine as whatever because the code is not littered with avoid_ub (yet), but eventually there may end up being a lot more, and I do think it would be bothersome to have to pay attention to ifdef/ifndef when just reading some code

If I'm the only one on this issue though I can give in

Reply:
I agree with prioritizing the original code first, but yeah I think we should stick to 1 block when possible, at least for now